futures-util/src/io/mod.rs

```diff
@@ -80,6 +122,40 @@ pub trait AsyncReadExt: AsyncRead {
     ///
     /// In the case of an error the buffer and the object will be discarded, with
     /// the error yielded.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(async_await, await_macro, futures_api)]
+    /// # futures::executor::block_on(async {
+    /// use futures::io::AsyncReadExt;
+    /// use std::io::Cursor;
+    ///
+    /// let mut reader = Cursor::new([1, 2, 3, 4]);
+    /// let mut output = [0u8; 4];
+    ///
+    /// await!(reader.read_exact(&mut output))?;
+    ///
+    /// assert_eq!(output, [1, 2, 3, 4]);
+    /// # Ok::<(), Box<std::error::Error>>(()) }).unwrap();
+    /// ```
+    ///
+    /// ## EOF is hit before `buf` is filled
```

Comment:
Maybe write it as "End of file". Some people might not know the term

Reply:
I expanded the earlier "EOF" in this doc-comment to "end of file (EOF)" and left it as shorthand in this heading.